api/src/main/java/org/apache/brooklyn/api/objs/BrooklynObject.java

```diff
@@ -141,4 +149,11 @@
          */
         boolean unsubscribe(SubscriptionHandle handle);
     }
+    
+    public interface RelationSupport<T extends BrooklynObject> {
```

Comment:
Cool -- I hadn't seen that!  Very nice for e.g. `policies()` and `tags()`.

However it's unclear what that should do in the case of *maps* like relations -- and config.  I note this isn't applied to `ConfigurationSupport`.  Suggest deferring this.  It can always be added once we know what its semantics should be.  /cc @aledsage